app/components/documents/document-card.js

```diff
@@ -202,8 +226,21 @@ export default class DocumentsDocumentCardComponent extends Component {
 
   @task
   *addPiece() {
+    if (this.signFlow) {
+      const status = yield this.signFlow.belongsTo('status').reload();
+      if (status.uri !== CONSTANTS.SIGNFLOW_STATUSES.MARKED) {
+        yield this.deleteUploadedPiece.perform();
+        this.router.refresh();
```

Comment:
A refresh from component? Should be the responsibility of controllers/routes.
Also, why are we refreshing here?
We could just do `deleteUploadedPiece` and show the warning.

Reply:
tested this locally and the only issue is that the actions dropdown stays visible after we reloaded the status.
So running `loadSignatureRelatedData` again is enough (only 1 card needs to be refreshed, not the entire route)